microsoft/testsuites/vm_extensions/custom_script.py

```diff
@@ -0,0 +1,533 @@
+        script_base64 = base64.b64encode(compressed_script).decode("utf-8")
+
+        blob_url = retrieve_storage_blob_url(
+            node=node,
+            environment=environment,
+            container_name=container_name,
+            blob_name=blob_name,
+            test_file=test_file,
+        )
+
+        settings = {"fileUris": [blob_url], "script": script_base64}
+
+        _create_and_verify_extension_run(
+            node=node,
+            settings=settings,
+            execute_commands=[CommandInfo(test_file, 0)],
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Custom Script VM extension with private Azure storage file uri
+        without a sas token.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_private_script_without_sas_run_failed(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        container_name = "cselisa"
+        blob_name = "cselisa.sh"
+        test_file = "/tmp/lisatest.txt"
+
+        blob_url = retrieve_storage_blob_url(
+            node=node,
+            environment=environment,
+            container_name=container_name,
+            blob_name=blob_name,
+            test_file=test_file,
+        )
+
+        settings = {
+            "fileUris": [blob_url],
+            "commandToExecute": f"sh {blob_name}",
+        }
+
+        # Expect HttpResponseError
+        _create_and_verify_extension_run(
+            node=node,
+            settings=settings,
+            assert_exception=HttpResponseError,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Custom Script VM extension with private Azure storage file uri
+        without a sas token but with storage account credentials.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_private_script_with_storage_credentials_run(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        container_name = "cselisa"
+        blob_name = "cselisa.sh"
+        test_file = "/tmp/lisatest.txt"
+
+        blob_url = retrieve_storage_blob_url(
+            node=node,
+            environment=environment,
+            container_name=container_name,
+            blob_name=blob_name,
+            test_file=test_file,
+        )
+
+        credentials = retrieve_storage_account_name_and_key(
+            node=node, environment=environment
+        )
+
+        settings = {"fileUris": [blob_url], "commandToExecute": f"sh {blob_name}"}
+
+        protected_settings = {
+            "storageAccountName": credentials["account_name"],
+            "storageAccountKey": credentials["account_key"],
+        }
+
```

Comment:
Make sure the key is not leaked in the log. If it is, add it to secret to mask it in log.